### PR TITLE
Stickier cyborg beakers

### PR DESCRIPTION
**Pull request by Crazylemon64**
The beakers that mediborgs have should now be less prone to getting caught in certain machines, or less capable of finding their way into the hands of some unfortunate doctor. I understand that the chem dispenser already checks if the character is a robot, but hopefully this should watch for other(?) NODROP cases, now, too.

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
without looking into the further details of this and really piecing it together (sorry, super tired) it seems like there would be a better way to handle this.

**Comment by Crazylemon64** [comment]
Hmm. Admittedly, I think I could move the "Stuck" message into the unequip part of the code, yeah, though I fear that might be a bit too much of a generalization. But I'm not sure how I could prevent the machines from taking the cyborg's beaker without directly touching their code, since the beakers are moved whether .drop_item() succeeds are not.

**Comment by Markolie** [comment]
I believe this is how most machinery handles it anyway.

File: code/game/dna/dna_modifier.dm [hunk 1]
```diff
@@ -217,10 +217,13 @@
 			return
 
 		beaker = item
-		user.drop_item()
-		item.loc = src
-		user.visible_message("[user] adds \a [item] to \the [src]!", "You add \a [item] to \the [src]!")
-		return
+		if(user.drop_item())
+			item.loc = src
```
Comment (PJB3005) [inline code/game/dna/dna_modifier.dm]: `forceMove()`

File: code/game/machinery/Sleeper.dm [hunk 2]
```diff
@@ -300,11 +300,13 @@
 	if(istype(G, /obj/item/weapon/reagent_containers/glass))
 		if(!beaker)
 			beaker = G
-			user.drop_item()
-			G.loc = src
-			user.visible_message("[user] adds \a [G] to \the [src]!", "You add \a [G] to \the [src]!")
-			src.updateUsrDialog()
-			return
+			if (user.drop_item())
+				G.loc = src
```
Comment (PJB3005) [inline code/game/machinery/Sleeper.dm]: `forceMove()`

File: code/game/machinery/constructable_frame.dm [hunk 3]
```diff
@@ -153,7 +153,7 @@
 			if(istype(P, /obj/item))
 				var/success
 				for(var/I in req_components)
-					if(istype(P, I) && (req_components[I] > 0))
+					if(istype(P, I) && (req_components[I] > 0) && !(P.flags & NODROP))
```
Comment (PJB3005) [inline code/game/machinery/constructable_frame.dm]: Can use the `^` operator for this.

File: code/game/machinery/cryo.dm [hunk 4]
```diff
@@ -279,9 +279,11 @@
 			return
 
 		beaker =  G
-		user.drop_item()
-		G.loc = src
-		user.visible_message("[user] adds \a [G] to \the [src]!", "You add \a [G] to \the [src]!")
+		if(user.drop_item())
+			G.loc = src
```
Comment (PJB3005) [inline code/game/machinery/cryo.dm]: `forceMove()`

File: code/modules/reagents/Chemistry-Machinery.dm [hunk 5]
```diff
@@ -218,15 +218,17 @@
...
-		B.loc = src
-		user << "You set [B] on the machine."
-		nanomanager.update_uis(src) // update all UIs attached to src
-		if(!icon_beaker)
-			icon_beaker = image('icons/obj/chemical.dmi', src, "disp_beaker") //randomize beaker overlay position.
-		icon_beaker.pixel_x = rand(-10,5)
-		overlays += icon_beaker
-		return
+		if(user.drop_item())
+			B.loc = src
```
Comment (PJB3005) [inline code/modules/reagents/Chemistry-Machinery.dm]: `forceMove()`

File: code/game/machinery/iv_drip.dm [hunk 6]
```diff
@@ -63,12 +63,14 @@
...
 			return
 
-		user.drop_item()
-		W.loc = src
-		src.beaker = W
-		user << "You attach \the [W] to \the [src]."
-		src.update_icon()
-		return
+		if(user.drop_item())
+			W.loc = src
```
Comment (PJB3005) [inline code/game/machinery/iv_drip.dm]: `forceMove()`

File: code/game/machinery/constructable_frame.dm (hunk 3 above)
Comment (Crazylemon64) [inline code/game/machinery/constructable_frame.dm]: As in, `P.flags ^ NODROP`?

Comment (Markolie) [inline code/game/machinery/constructable_frame.dm]: Honestly, I'd just stick with this. It's much more readable and it doesn't matter much.

**Comment by Crazylemon64** [comment]
Actually: The change to machine frames, as it stands now, will prevent drones or engineering cyborgs from adding cabling to machine frames. Should I modify the code to let stacks through even if they're `NODROP`?

**Comment by Markolie** [comment]
Please do.

**Pull request merged**